File: turboaffiliate/static/javascript/json.js

var JSON = {
  departamento : function(url)
  {
    $.get(url+'/departamentos?tg_format=json', function(data)
    {
      $.each(data.departamentos, function(i, departamento)
      {
        var option = $('<option />');
        option.val(departamento.id);
        option.text(departamento.nombre);
        $('.departamento').append(option);
      });
    });
  },
  
  municipio : function(url, departamento, municipios)
  {
    $.get(url + '/municipios/' + departamento + '?tg_format=json', function(data)
    {
      var select = $(municipios);
      select.empty();
      var index;
      for(index = 0; index < data.municipios.length; index++) {
        var municipio = data.municipios[index];
        select.append($('<option />', {
          value: municipio.id,
          text : municipio.nombre
        }));
      }
      select.trigger('change', true);
    });
  },
  
  municipioE : function(url, departamento, municipios)
  {
    $.get(url + '/municipios/' + departamento + '?tg_format=json', function(data)
    {
      $.each(data.municipios, function(i, municipio)
      {
        var option = $('<option />');
        option.val(municipio.id);
        option.text(municipio.nombre);
        $(municipios).append(option);
      });
    });
  },
  
  cotizacion : function(url)
  {
    $.get(url + '/cotizacion?tg_format=json',function(data)
    {
      $.each(data.cotizaciones, function(i, cotizacion)
      {
        var option = $('<option />');
        option.val(cotizacion.id);
        option.text(cotizacion.nombre)
        $('.cotizaciones').append(option);
      });
    });
  },
  bancos : function(url)
  {
    $.get(url + '/bancos?tg_format=json', function(data)
    {
      $.each(data.bancos, function(i, banco)
      {
        var option = $('<option />');
        option.val(banco.id);
        option.text(banco.nombre)
        $('.bancos').append(option);
      });
    });
  },
  cuentas : function(url)
  {
    $.get(url + '/cuentas?tg_format=json', function(data)
    {
      $.each(data.cuentas, function(i, cuenta)
      {
        var option = $('<option />');
        option.val(cuenta.id);
        option.text(cuenta.id + ' ' + cuenta.name)
        $('.cuentas').append(option);
      });
    });
  },
  jubilados : function(url)
  {
    $.get(url + '/jubilados?tg_format=json', function(data)
    {
      $.each(data.cotizaciones, function(i, cotizacion)
      {
        var option = $('<option />');
        option.val(cotizacion.id);
        option.text(cotizacion.nombre)
        $('.jubilados').append(option);
      });
    });
  }
}

File: turboaffiliate/controllers/json.py
# -*- coding: utf8 -*-
#
# json.py
# This file is part of TurboAffiliate
#
# Copyright (c) 2010, 2011 Carlos Flores <cafg10@gmail.com>
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.

from turbogears import controllers, expose, validate, validators
from turboaffiliate import model

class JSON(controllers.Controller):
    
    @expose('json')
    def departamentos(self):
        
        """Regresa una lista de departamentos en formato JSON"""
        
        return dict(departamentos=model.Departamento.select())
    
    @expose('json')
    @validate(validators=dict(departamento=validators.Int()))
    def municipios(self, departamento):
        
        """Regresa una lista de municipios en fromato JSON"""
        
        departamento = model.Departamento.get(departamento)
        
        return dict(municipios=model.Municipio.selectBy(departamento=departamento))
    
    @expose('json')
    def cuentas(self):
        
        """Muestra una lista de las cuentas del sistema"""
        
        return dict(cuentas=model.Account.select())
    
    @expose('json')
    def cotizacion(self):
        
        """Muestra una lista de los métodos de cotización"""
        
        return dict(cotizaciones=model.Cotizacion.select())
    
    @expose('json')
    def bancos(self):
        
        """Muestra una lista con todos los bancos disponibles"""
        
        return dict(bancos=model.Banco.selectBy(depositable=True))
    
    @expose('json')
    def jubilados(self):
        
        return dict(cotizaciones=model.Cotizacion.selectBy(jubilados=True))
